parsimonious/tests/test_expressions.py

```diff
@@ -289,7 +289,27 @@ def test_unicode_surrounding_parens(self):
 
         """
         eq_(text_type(Grammar('foo = ("foo" ("bar" "baz"))')),
-            u'foo = "foo" ("bar" "baz")')
+            u"foo = 'foo' ('bar' 'baz')")
+
+    def test_repr_special_character_escaping(self):
+        """Make sure special characters are properly escaped in the repr.
+
+        """
+
+        # backslash
+        eq_(repr(Grammar(r'''foo = "\\" ''')), u'''Grammar("foo = '\\\\\\\\'")''')
+
+        # single quote
+        eq_(repr(Grammar(r'''foo = "'" ''')), u'''Grammar('foo = "\\'"')''')
+
+        # escaped single quote inside a single quoted string
+        eq_(repr(Grammar(r'''foo = '\'' ''')), u'''Grammar('foo = "\\'"')''')
+
+        # double quote
+        eq_(repr(Grammar(r'''foo = '"' ''')), u'''Grammar('foo = \\'"\\'')''')
```

Comment:
The ones of these I've spot checked appear right, but boy are they mind-bending! We could prove they are right, without doing much extra work, by round-tripping them in the test and making sure the results are equal. For example…

```python
g = Grammar(...)
eq_(g, eval(repr(g)))
```

I believe all the `__eq__` methods I've defined in expressions.py should do the heavy lifting. No need to block this PR on this, though—just something to think about for later.